sound/soc/intel/boards/sof_board_helpers.h

```diff
@@ -84,6 +85,7 @@ struct sof_card_private {
 	struct snd_soc_dai_link *amp_link;
 
 	unsigned long link_order_overwrite;
+	unsigned long link_id_overwrite;
```

Comment:
I am not following what the 'link_id_overwrite' means. Is this supposed to be used for a single link, or for multple links?

It could be a naming convention issue, I am just not following the intended behavior

Reply:
We use this to overwrite the id field of all DAI links. Like the macros defined in sof_ssp_amp.c

#define SSP_AMP_LINK_ORDER	SOF_LINK_ORDER(SOF_LINK_HDMI_IN, \
					SOF_LINK_AMP,            \
					SOF_LINK_DMIC01,         \
					SOF_LINK_DMIC16K,        \
					SOF_LINK_IDISP_HDMI,     \
					SOF_LINK_BT_OFFLOAD,     \
					SOF_LINK_NONE)
=> this is the dai link order, HDMI_IN is the first link, BT_OFFLOAD is the last.

#define SSP_AMP_LINK_IDS	SOF_LINK_ORDER(HDMI_IN_BE_ID, \
					SPK_BE_ID,            \
					DMIC01_BE_ID,         \
					DMIC16K_BE_ID,        \
					INTEL_HDMI_BE_ID,     \
					BT_OFFLOAD_BE_ID,     \
					0)
=> this is the id for each dai link

Reply:
if the 'overwrite' a boolean or can it contain a value > 1?

Reply:
This variable is an encoded data. When generating DAI link array, we will extract the id of each link from this variable if not zero.